src/build/__init__.py

```diff
@@ -161,6 +163,17 @@ def __init__(self, srcdir='.', config_settings=None, python_executable=sys.execu
             python_executable=python_executable,
         )
 
+    @property
+    def python_executable(self):  # type: () -> Union[bytes, Text]
+        # make mypy happy
+        if not isinstance(self.hook.python_executable, typing.Text) and not isinstance(self.hook.python_executable, bytes):
+            raise TypeError('python_executable is not a string: {}'.format(type(self.hook.python_executable)))
+        return self.hook.python_executable
+
+    @python_executable.setter
+    def python_executable(self, value):  # type: (Union[bytes, Text]) -> None
+        self.hook.python_executable = value
```

Comment:
Shouldn't the type  validation be performed here, and not during read?

Reply:
It was to make mypy happy. We have type hints here.

Reply:
I think we should raise an error if the input is not of those types.

Reply:
The rest of the API doesn't perform any kind of input validation - why single out `python_executable`?